### DISCUSSION TITLE
Markdown > HTML: How to keep fenced DIVs from receiving heading classes?

### DISCUSSION THREAD

**Discussion by PlainMartin** [jgm/pandoc#9966] [Q&A] [answered]
This Markdown source:

~~~
::: {.class1}

# Hello world {.class2}

## Hello again {.class3}

:::
~~~

will be converted to:

~~~
<section id="hello-world" class="class2 class1">
<h1 class="class2">Hello world</h1>
<h2 class="class3" id="hello-again">Hello again</h2>
</section>
~~~

So I get HTML _sections_ with the H1 class(es) and ID added, which is not what I want.

I tried adding `--section-divs=false` to the command line, but I get the same result.

How could I get

~~~
<div class="class1">
<h1 class="class2" id="hello-world">Hello world</h1>
<h2 class="class3" id="hello-again">Hello again</h2>
</div>
~~~

Thank you.

_Edit: I found that adding a paragraph (even with a single character) right after the fenced DIV marker will give the desired result, but I would still like to know if it can be achieved with a head1 right after the fenced DIV marker._

**Comment by tarleb** [comment, marked as answer]
Using an (possibly empty) HTML comment before the first header should give the desired result:

``` markdown
::: {.class1}

<!-- -->

# Hello world {.class2}

## Hello again {.class3}

:::
```

**Comment by PlainMartin** [comment]
Thank you! 

So, just for context – this is not considered a bug, but intended functionality? Because I understand when elements will _inherit_ rules, but it gets a bit confusing when they are added to their _parents_ (and as I said, `--section-divs=false` doesn’t seem to affect this).

**Reply by tarleb** [reply]
It's complicated and remains an open issue. See #5965.
